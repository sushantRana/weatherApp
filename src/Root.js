import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootreducer from './reducers/rootreducer';

export default ({ children, initialState = {} }) => {
  const store = createStore(
    rootreducer,
    initialState,
  );

  return (
    <Provider store={store}>
      {children}
    </Provider>
  );
}