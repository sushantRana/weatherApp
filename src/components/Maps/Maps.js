import React from 'react';
import { withScriptjs, withGoogleMap, GoogleMap, Marker } from "react-google-maps"

export class Mapsfunction extends React.Component {
    render() {
      const { lat, lng, isMarkerShown, onMarkerClick } = this.props;
      return (
        <GoogleMap
          defaultZoom={8}
          defaultCenter={{ lat: lat, lng: lng }}
          center={{ lat: lat, lng: lng }}
        >
          {isMarkerShown && 
          	<Marker 
          		position={{ lat: lat, lng: lng }} 
          		onClick={() => onMarkerClick(lat, lng)}
          	/>
          }
        </GoogleMap>
      )
    }
};

const Maps = withScriptjs(withGoogleMap(Mapsfunction));

export default Maps;