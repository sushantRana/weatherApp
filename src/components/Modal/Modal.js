import Modal from 'react-modal';
import React from 'react';
import ChartsOfWeather from '../../containers/ChartsOfWeather/ChartsOfWeather';

const customStyles = {
  content : {
    top                   : '50%',
    left                  : '50%',
    right                 : 'auto',
    bottom                : 'auto',
    marginRight           : '-50%',
    height                : '400px',
    width                 : '60%',
    transform             : 'translate(-50%, -50%)'
  }
};

if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#root')
class ModalComponent extends React.Component {
  render() {
    return <Modal
      isOpen={this.props.isOpen}
      onAfterOpen={this.afterOpenModal}
      onRequestClose={this.props.closeModal}
      style={customStyles}
      contentLabel="Example Modal"
    >
      <ChartsOfWeather 
        lat={this.props.lat}
        lng={this.props.lng}
      />
    </Modal>
  }
}
export default ModalComponent;