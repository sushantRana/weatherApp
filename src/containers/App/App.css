.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

body {
  font-family: Helvetica, sans-serif;
}
.headerDiv {
  height: 3em;
  background: #302FB0;
  display: flex;
  align-items: center;
  justify-content: center;
}
.title {
  margin: 0;
  color: #fff;
}
.location-search-input {  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.16), 0 0 0 1px rgba(0,0,0,0.08);
    border: honeydew;
    display: block;
    width: 100%;
    padding: 16px;
    font-size: 16px;
    border-radius: 2px;
    outline: none;
  }

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
