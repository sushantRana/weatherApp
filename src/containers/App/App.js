import React from 'react';
import AutocompleteBox from '../AutocompleteBox/AutocompleteBox';
import Modal from '../../components/Modal/Modal';
import Maps from '../../components/Maps/Maps';
import { GoogleURL } from '../../config/config';
import { connect } from 'react-redux';

import './App.css';

export class App extends React.Component {
  state = {};

  handleMarkerClick = (lat, lng) => {
    this.setState({modalIsOpen : true});
  }
  closeModal = () => {
    this.setState({modalIsOpen : false}); 
  }

  render() {
    return (
      <div>
        <Modal
          isOpen={this.state.modalIsOpen}
          closeModal={this.closeModal}
          lat={this.props.lat}
          lng={this.props.lng}
        />
        <div className='headerDiv'>
          <h3 className='title'>Weather Prediction App</h3>
        </div>
        <AutocompleteBox />
        <Maps
          onMarkerClick={this.handleMarkerClick}
          isMarkerShown
          googleMapURL={GoogleURL}
          loadingElement={<div style={{ height: `100%` }} />}
          containerElement={<div style={{ height: `800px` }} />}
          mapElement={<div style={{ height: `100%` }} />}
          lat={this.props.lat}
          lng={this.props.lng}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  lat: state.weather.lat,
  lng: state.weather.lng
});

export default connect(mapStateToProps)(App)
