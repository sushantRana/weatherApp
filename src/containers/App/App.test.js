import React from 'react';
import ReactDOM from 'react-dom';
import { App } from './App';
import Root from '../../Root';
import Maps from '../../components/Maps/Maps';
import Modal from '../../components/Modal/Modal';

import { mount } from 'enzyme';
import moxios from 'moxios';

import { shallow } from 'enzyme';

describe('App', () => {
  const props = {
	lat: 19.0759837,
	lng: 72.8776559,
    weather: {}
}
  const wrapped = mount(
	  <Root>
	    <App />
	  </Root>
	);
  const app = shallow(<App {...props} />);
  it ('renders without crashing', () => {
    expect(app).toMatchSnapshot();
  });

  it('Shows a Maps', () => {
    expect(app.find(Maps).length).toEqual(1);
  });
  it('can fetch a list of places and display them on map', (done) => {
	  wrapped.find('PlacesAutocomplete').simulate('select', 'Bangalore');
	  moxios.wait(() => {
	      wrapped.update();
	      expect(wrapped.find(Maps).length).toEqual(1);
	      done();
	      wrapped.unmount();
	  });
   });
});
