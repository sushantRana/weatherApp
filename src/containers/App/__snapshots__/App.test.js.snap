// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    lat={19.0759837}
    lng={72.8776559}
    weather={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ModalComponent
          closeModal={[Function]}
          isOpen={undefined}
          lat={19.0759837}
          lng={72.8776559}
/>,
        <div
          className="headerDiv"
>
          <h3
                    className="title"
          >
                    Weather Prediction App
          </h3>
</div>,
        <Connect(AutocompleteBox) />,
        <withScriptjs(withGoogleMap(Mapsfunction))
          containerElement={
                    <div
                              style={
                                        Object {
                                                  "height": "800px",
                                                }
                              }
                    />
          }
          googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyC4R6AN7SmujjPUIGKdyao2Kqitzr1kiRg&v=3.exp&libraries=geometry,drawing,places"
          isMarkerShown={true}
          lat={19.0759837}
          lng={72.8776559}
          loadingElement={
                    <div
                              style={
                                        Object {
                                                  "height": "100%",
                                                }
                              }
                    />
          }
          mapElement={
                    <div
                              style={
                                        Object {
                                                  "height": "100%",
                                                }
                              }
                    />
          }
          onMarkerClick={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeModal": [Function],
          "isOpen": undefined,
          "lat": 19.0759837,
          "lng": 72.8776559,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3
            className="title"
>
            Weather Prediction App
</h3>,
          "className": "headerDiv",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Weather Prediction App",
            "className": "title",
          },
          "ref": null,
          "rendered": "Weather Prediction App",
          "type": "h3",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "containerElement": <div
            style={
                        Object {
                                    "height": "800px",
                                  }
            }
/>,
          "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyC4R6AN7SmujjPUIGKdyao2Kqitzr1kiRg&v=3.exp&libraries=geometry,drawing,places",
          "isMarkerShown": true,
          "lat": 19.0759837,
          "lng": 72.8776559,
          "loadingElement": <div
            style={
                        Object {
                                    "height": "100%",
                                  }
            }
/>,
          "mapElement": <div
            style={
                        Object {
                                    "height": "100%",
                                  }
            }
/>,
          "onMarkerClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ModalComponent
            closeModal={[Function]}
            isOpen={undefined}
            lat={19.0759837}
            lng={72.8776559}
/>,
          <div
            className="headerDiv"
>
            <h3
                        className="title"
            >
                        Weather Prediction App
            </h3>
</div>,
          <Connect(AutocompleteBox) />,
          <withScriptjs(withGoogleMap(Mapsfunction))
            containerElement={
                        <div
                                    style={
                                                Object {
                                                            "height": "800px",
                                                          }
                                    }
                        />
            }
            googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyC4R6AN7SmujjPUIGKdyao2Kqitzr1kiRg&v=3.exp&libraries=geometry,drawing,places"
            isMarkerShown={true}
            lat={19.0759837}
            lng={72.8776559}
            loadingElement={
                        <div
                                    style={
                                                Object {
                                                            "height": "100%",
                                                          }
                                    }
                        />
            }
            mapElement={
                        <div
                                    style={
                                                Object {
                                                            "height": "100%",
                                                          }
                                    }
                        />
            }
            onMarkerClick={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "closeModal": [Function],
            "isOpen": undefined,
            "lat": 19.0759837,
            "lng": 72.8776559,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3
              className="title"
>
              Weather Prediction App
</h3>,
            "className": "headerDiv",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Weather Prediction App",
              "className": "title",
            },
            "ref": null,
            "rendered": "Weather Prediction App",
            "type": "h3",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "containerElement": <div
              style={
                            Object {
                                          "height": "800px",
                                        }
              }
/>,
            "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyC4R6AN7SmujjPUIGKdyao2Kqitzr1kiRg&v=3.exp&libraries=geometry,drawing,places",
            "isMarkerShown": true,
            "lat": 19.0759837,
            "lng": 72.8776559,
            "loadingElement": <div
              style={
                            Object {
                                          "height": "100%",
                                        }
              }
/>,
            "mapElement": <div
              style={
                            Object {
                                          "height": "100%",
                                        }
              }
/>,
            "onMarkerClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
